pkg/kapp/diff/change.go

```diff
@@ -39,13 +40,16 @@ type ChangeImpl struct {
 	// appliedRes is an unmodified copy of what's being applied
 	appliedRes ctlres.Resource
 
+	// clusterOriginalRes is an unmodified copy what's present on the cluster
+	clusterOriginalRes ctlres.Resource
+
 	configurableTextDiff *ConfigurableTextDiff
 	opsDiff              *OpsDiff
 }
 
 var _ Change = &ChangeImpl{}
 
-func NewChange(existingRes, newRes, appliedRes ctlres.Resource) *ChangeImpl {
+func NewChange(existingRes, newRes, appliedRes, clusterOriginalRes ctlres.Resource) *ChangeImpl {
```

Comment:
Why are we doing deep copies of the resources here? Is this something we should do for clusterOriginalRes as well?
Or is the intended effect of deep copying ensured?

Reply:
(Look at line 58)

Reply:
Hmm, good question, I will try to see why we are doing it for others.